version: '3.9'

services:
  postgres:
    image: postgres
    restart: always
    environment:
      - POSTGRES_DB=db_farm_2023
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=1123QwER
      - PGDATA=/var/lib/postgresql/data/pgdata
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data

  web:
    image: sadidocker/farm_app
    build: .
    entrypoint: [ "./wait-for-it.sh", "postgres:5432", "--timeout=60", "--" ]
    command: python manage.py runserver 0.0.0.0:8000
    ports:
      - "8000:8000"
    env_file:
      - .env
    volumes:
      - .:/code
      - ./staticfiles:/tmp/app-resources/staticfiles
      - media_data:/app/media/uploads
    depends_on:
      - postgres
    environment:
      - DATABASE_URL=postgres://postgres:1123QwER@postgres:5432/db_farm_2023

  nginx:
    image: nginx:latest
    ports:
      - "81:80"
      - "443:443"
    volumes:
      - ./static:/home/app/web/staticfiles
      - ./nginx/conf.d:/etc/nginx/conf.d
      - ./nginx/logs:/var/log/nginx
      - ./certbot/www:/var/www/certbot/:ro
      - ./certbot/conf/:/etc/nginx/ssl/:ro
      - ./nginx/conf.d/web.conf:/etc/nginx/conf.d/web.conf:ro
    depends_on:
      - web

volumes:
  postgres_data:
  media_data:
